c_src/filewatch_inotify.c

```diff
@@ -186,36 +208,43 @@ static void send_output(ErlDrvData self_)
 
     buf[pos++] = ERL_DRV_PORT;
     buf[pos++] = driver_mk_port(self->port);
-
-    struct message *msg;
-    for(uint i = 0; i < self->n_events; ++i) {
-        msg = &(self->msgs[i]);
+    
+    //Remove all elements from abstab and encode each into buf
+    struct Message msg;
+    int num_entries_removed = 0;
+    while(remove_next_absTab(self->abstab, &msg)) {
         buf[pos++] = ERL_DRV_INT;
-        buf[pos++] = msg->wd;
+        buf[pos++] = msg.wd;
         buf[pos++] = ERL_DRV_STRING;
-        buf[pos++] = (ErlDrvTermData) msg->name;
-        buf[pos++] = strnlen(msg->name, NAME_MAX);
+        buf[pos++] = (ErlDrvTermData) msg.name;
+        buf[pos++] = strnlen(msg.name, NAME_MAX);
```

Comment:
True.

My main worry is that this might accidentally copy extra bytes beyond `msg.name[msg.len]` if there is no null byte at `msg.name[msg.len]`. However, `strncpy` does say it copies a null byte, so maybe I'm just being paranoid.